server/etcdserver/api/v3rpc/health.go

```diff
@@ -57,7 +61,10 @@ func (hc *healthChecker) StartServe() {
 	hc.hs.SetServingStatus(allGRPCServices, healthpb.HealthCheckResponse_SERVING)
 }
 
-func (hc *healthChecker) StopServe(reason string) {
+func (hc *healthNotifier) stopServe(reason string) {
+	if !hc.stopGRPCServiceOnDefrag {
```

Comment:
This condition would make more sense if it was in `defragFinished`.

Reply:
Done. I guess you meant `defragStarted` :p